spec/ruby/core/io/set_encoding_by_bom_spec.rb

```diff
@@ -67,5 +67,11 @@
 
       -> { @io.set_encoding_by_bom }.should raise_error(ArgumentError, 'encoding is set to UTF-8 already')
     end
+
+    it 'returns exception if encoding conversion is already set' do
+      @io.set_encoding(nil, Encoding::UTF_8)
```

Comment:
This fails on CRuby 2.7, could you adapt the spec so it works on CRuby too?
https://github.com/oracle/truffleruby/pull/2372/checks?check_run_id=2765087425

Reply:
Should be good now! Although it means that there's an incompatibility in the API where `nil` is a valid first argument here for `#set_encoding`.